packages/squiggle-lang/src/value/BaseValue.ts

```diff
@@ -45,4 +51,21 @@ export abstract class BaseValue {
     const argsStr = `{${this.tags.toString()}}`;
     return `${valueString}, with tags ${argsStr}`;
   }
+
+  abstract serializePayload(
+    visitor: SquiggleSerializationVisitor
+  ): SerializedPayload;
```

Comment:
Note that all this stuff about `serializePayload` vs `serialize` is one level below the main serialization framework in `src/serialization/*`.

Serialization framework doesn't know anything about values being discriminated unions.

Instead, it just calls `serializeValue` and `deserializeValue`; those are specific to `Value` type.